--- gcc/config/i386/i386.c.orig	2011-04-29 10:43:05.717344059 +0800
+++ gcc/config/i386/i386.c	2011-04-29 10:45:47.188234204 +0800
@@ -2329,7 +2329,7 @@
 enum tls_dialect ix86_tls_dialect = TLS_DIALECT_GNU;
 
 /* Which unit we are generating floating point math for.  */
-enum fpmath_unit ix86_fpmath;
+enum fpmath_unit ix86_fpmath = FPMATH_SSE;
 
 /* Which cpu are we scheduling for.  */
 enum attr_cpu ix86_schedule;
@@ -2477,7 +2477,7 @@
 #endif
 
 /* Whether -mtune= or -march= were specified */
-static int ix86_tune_defaulted;
+static int ix86_tune_defaulted = 1;
 static int ix86_arch_specified;
 
 /* A mask of ix86_isa_flags that includes bit X if X
@@ -3489,6 +3489,7 @@
   if (TARGET_MACHO && TARGET_64BIT)
     flag_pic = 2;
 
+  ix86_tune_string = "atom";
   /* Need to check -mtune=generic first.  */
   if (ix86_tune_string)
     {
@@ -3564,6 +3565,7 @@
 	       ix86_stringop_string, prefix, suffix, sw);
     }
 
+  ix86_arch_string = "core2";
   if (!ix86_arch_string)
     ix86_arch_string = TARGET_64BIT ? "x86-64" : SUBTARGET32_DEFAULT_CPU;
   else
@@ -4073,7 +4075,7 @@
       && ! TARGET_SSE)
     error ("%ssseregparm%s used without SSE enabled", prefix, suffix);
 
-  ix86_fpmath = TARGET_FPMATH_DEFAULT;
+  ix86_fpmath = FPMATH_SSE;
   if (ix86_fpmath_string != 0)
     {
       if (! strcmp (ix86_fpmath_string, "387"))
