--- gcc-4.6.1/boehm-gc/include/private/gcconfig.h	2011-11-26 20:42:33.061763028 +0100
+++ gcc-4.6.1-boehm-mips/boehm-gc/include/private/gcconfig.h	2011-11-26 20:46:24.756983800 +0100
@@ -1380,7 +1380,8 @@
 #     endif
 #     define USE_GENERIC_PUSH_REGS
 #     if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 2 || __GLIBC__ > 2
-#        define LINUX_STACKBOTTOM
+#warning Changing to HEURISTIC2 because of QEMU
+#        define HEURISTIC2
 #     else
 #        define STACKBOTTOM 0x80000000
 #     endif
