--- a/gcc/config/i386/t-linux64.orig	2013-05-15 05:31:20.000000000 -0700
+++ b/gcc/config/i386/t-linux64	2013-06-11 14:11:15.188517900 -0700
@@ -25,7 +25,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64$(call if_multiarch,:x86_64-linux-gnu) \
+MULTILIB_OSDIRNAMES = ../lib$(call if_multiarch,:x86_64-linux-gnu) \
 	$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)$(call if_multiarch,:i386-linux-gnu)
 
 LIBGCC = stmp-multilib
--- a/gcc/config/i386/linux64.h.orig	2013-05-15 05:31:20.000000000 -0700
+++ b/gcc/config/i386/linux64.h	2013-06-11 14:11:56.196515316 -0700
@@ -63,7 +63,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
 
 #if TARGET_64BIT_DEFAULT
 #define SPEC_32 "m32"
